osquery/logger/plugins/buffered.cpp

```diff
@@ -67,14 +73,53 @@ void BufferedLogForwarder::check() {
       VLOG(1) << "Error sending status to logger: " << status.getMessage();
     } else {
       // Clear the status logs once they were sent.
-      iterate(indexes, ([&results, this](std::string& index) {
+      iterate(indexes, ([this](std::string& index) {
                 if (!isStatusIndex(index)) {
                   return;
                 }
                 deleteDatabaseValue(kLogs, index);
               }));
     }
   }
+
+  // Purge any logs exceeding the max after our send attempt
+  if (FLAGS_buffered_log_max > 0) {
+    purge();
+  }
+}
+
+void BufferedLogForwarder::purge() {
+  std::vector<std::string> indexes;
+  auto status = scanDatabaseKeys(kLogs, indexes, index_name_, 0);
```

Comment:
We will have to be careful about letting this count get out of sync.

What do you think of storing the count in-memory (generating it through a scan on startup)? Then we can do our locking around quick variable updates rather than slower DB writes. It also allows a chance for recovery on restart if the count does go out of sync.